### ISSUE TITLE
'The super constructor to "inherits" must not be null or undefined' exception

### ISSUE THREAD

**Issue by phlummox** [xBrowserSync/API#1]
Hi - attempting to run with node version 8.9.4 and mongodb version 2.6.10 on Ubuntu 16.04, an exception got thrown when I attempted to start the API server:

~~~
> sudo node api.js
util.js:969
    throw new TypeError('The super constructor to "inherits" must not ' +
    ^

TypeError: The super constructor to "inherits" must not be null or undefined
    at Object.inherits (util.js:969:11)
    at Object.<anonymous> (.../node_modules/spdy/lib/spdy/connection.js:86:6)
    at Module._compile (module.js:643:30)
    at Object.Module._extensions..js (module.js:654:10)
    at Module.load (module.js:556:32)
    at tryModuleLoad (module.js:499:12)
    at Function.Module._load (module.js:491:3)
    at Module.require (module.js:587:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (.../node_modules/spdy/lib/spdy.js:20:19)
~~~

I found that changing the version of `restify` in the dependencies in `package.json` from "4.0.x" to "4.3.x" (as suggested in [this](https://github.com/slashdotdash/node-ledger-rest/issues/6) issue report), and re-building, got the service at least starting up:

~~~
diff --git a/package.json b/package.json
index 00b671e..fae0c16 100644
--- a/package.json
+++ b/package.json
@@ -20,7 +20,7 @@
     "mongojs": "2.4.x",
     "node-uuid": "1.4.x",
     "q": "1.4.x",
-    "restify": "4.0.x",
+    "restify": "4.3.x",
     "request": "2.73.x"
   },
   "devDependencies": {
~~~

I still don't seem to be getting any response when I  use `wget` to try accessing the API, nor are any log messages getting written, but I will investigate that further.

**Comment by nero120** [comment]
Thanks for raising this @phlummox, I'll be able to pick this up within the next few weeks. I'm having to do a lot of extra hours at work at the moment so I don't have any spare time. Though once that's out of the way I've got a new release for xBrowserSync due so I'll pick this issue up at the same time.

**Comment by phlummox** [comment]
No worries, thanks for that. Out of interest, what versions of node and mongodb have you been using? I'm just about to try building & running on a docker instance using node v7.7.2, as I see that's what's being used in the Travis build. cheers!

**Comment by EvgeniBaldzhiyski** [comment]
I have solution. The restify use old version of **spdy**. You need to update this package to latest version (**spdy@3.4.7** in this moment) and then remove **/path-to/node_modules/restify/node_modules/spdy**
I hope this will help you because I spent a lot time before cought this sht. 
Enjoy

**Comment by nero120** [comment]
@EvgeniBaldzhiyski wow that's awesome, good catch and thanks very much. I'm going to work on the api this weekend, upgrade the dependencies and convert to using typescript and webpack. I'll encorporate you fix into this release. Thanks again bud.

**Comment by nero120** [comment]
@phlummox sorry I missed your last message, I've just rebuilt my local environment after moving from ubuntu back to win10 (sad times). But the live server uses node v4.2.6 and mongodb v3.2.17.

I'll sort out a release that works using latest dependencies, and as I said above I then want to convert to TypeScript and webpack. Bit behind the times here!

**Comment by nero120** [comment]
Fixed in ed99fc4888f78db8a155b83de27058194145088f.
